main.nf

```diff
@@ -191,67 +235,655 @@ process get_software_versions {
+  set val(setname), val('decoy'), file('dmzidperco') into dmzidtsv_perco
+
+  script:
+  """
+  perco_to_tsv.py -p $perco --plates ${platenames.join(' ')} --fractions ${fractions.join(' ')}
+  """
+}
+
+// Collect percolator data of target/decoy and feed into PSM table creation
+tmzidtsv_perco
+  .concat(dmzidtsv_perco)
+  .groupTuple(by: 1)
+  .combine(quant_lookup)
+  .set { prepsm }
+
+// Set strips to false if not running hirief
+if (params.hirief) {
+  strips_for_deltapi
+    .map { it -> [it, trainingpep] }
+    .set { stripannot }
+} else {
+  strips_for_deltapi
+    .map { it -> [it, false, false]}
+    .set { stripannot }
+}
+
+
+/*
+* Step 3: Post-process peptide identification data
+*/
+
+process createPSMTable {
+
+  publishDir "${params.outdir}", mode: 'copy', overwrite: true, saveAs: {["target_psmlookup.sql", "target_psmtable.txt", "decoy_psmtable.txt"].contains(it) ? it : null}
+
+  input:
+  set val(setnames), val(td), file('psms?'), file('lookup') from prepsm
+  set file(tdb), file(ddb) from searchdbs
+  set val(allstrips), file(trainingpep) from stripannot
+
+  output:
+  set val(td), file("${td}_psmtable.txt") into psm_result
+  set val(td), file({setnames.collect() { it + '.tsv' }}) into setpsmtables
+  set val(td), file("${td}_psmlookup.sql") into psmlookup
+
+  script:
+  """
+  msspsmtable merge -i psms* -o psms.txt
+  msspsmtable conffilt -i psms.txt -o filtpsm --confidence-better lower --confidence-lvl 0.01 --confcolpattern 'PSM q-value'
+  msspsmtable conffilt -i filtpsm -o filtpep --confidence-better lower --confidence-lvl 0.01 --confcolpattern 'peptide q-value'
+  cp lookup psmlookup
+  msslookup psms -i filtpep --dbfile psmlookup ${params.onlypeptides ? '' : "--fasta ${td == 'target' ? tdb : "${ddb} --decoy"}"} ${params.martmap ? "--map ${martmap}" : ''}
+  msspsmtable specdata -i filtpep --dbfile psmlookup -o prepsms.txt
+  ${!params.noquant ? "msspsmtable quant -i prepsms.txt -o qpsms.txt --dbfile psmlookup --precursor ${params.isobaric && td=='target' ? '--isobaric' : ''}" : 'mv prepsms.txt qpsms.txt'}
+  sed 's/\\#SpecFile/SpectraFile/' -i qpsms.txt
+  ${!params.onlypeptides ? "msspsmtable genes -i qpsms.txt -o gpsms --dbfile psmlookup" : ''}
+  ${!params.onlypeptides ? "msslookup proteingroup -i qpsms.txt --dbfile psmlookup" : ''}
+  ${!params.onlypeptides ? "msspsmtable proteingroup -i gpsms -o pgpsms --dbfile psmlookup" : 'mv qpsms.txt pgpsms'}
+  ${params.hirief ? "peptide_pi_annotator.py -i $trainingpep -p pgpsms --o dppsms --stripcolpattern Strip --pepcolpattern Peptide --fraccolpattern Fraction --strippatterns ${allstrips.join(' ')} --intercepts ${allstrips.collect() { params.strips[it].intercept}.join(' ')} --widths ${allstrips.collect() { params.strips[it].fr_width}.join(' ')} --ignoremods \'*\'" : ''}
+  msspsmtable split -i ${params.hirief ? 'dppsms' : 'pgpsms'} --bioset
+  mv ${params.hirief ? 'dppsms' : 'pgpsms'} ${td}_psmtable.txt
+  mv psmlookup ${td}_psmlookup.sql
```

Comment:
You could use nextflows file renaming functionality to put this into a channel with a different name, thus not needing this...